app/src/org/commcare/views/media/MediaLayout.java

```diff
@@ -321,30 +369,28 @@ public void onPrepared(MediaPlayer mediaPlayer) {
                 videoView.setId(INLINE_VIDEO_PANE_ID);
                 return videoView;
             }
-
         } catch (InvalidReferenceException ire) {
-            Log.e(t, "invalid video reference exception");
+            Log.e(TAG, "invalid video reference exception");
             ire.printStackTrace();
-            error = "Invalid reference: " + ire.getReferenceString();
+            return getMissingImageView("Invalid reference: " + ire.getReferenceString());
         }
+    }
 
-        if (error != null) {
-            mMissingImage = new TextView(getContext());
-            mMissingImage.setText(error);
-            mMissingImage.setPadding(10, 10, 10, 10);
-            mMissingImage.setId(234873453);
-            return mMissingImage;
-        } else {
-            return null;
-        }
+    private TextView getMissingImageView(String errorMessage) {
+        missingImageText = new TextView(getContext());
+        missingImageText.setText(errorMessage);
+        missingImageText.setPadding(10, 10, 10, 10);
+        missingImageText.setId(MISSING_IMAGE_ID);
+        return missingImageText;
     }
 
     private boolean useResizingImageView() {
         // only allow ResizingImageView to be used if not also using smart inflation
         return !CommCarePreferences.isSmartInflationEnabled() &&
-                (ResizingImageView.resizeMethod.equals("full") ||
-                        ResizingImageView.resizeMethod.equals("half") ||
-                        ResizingImageView.resizeMethod.equals("width"));
+                (ResizingImageView.resizeMethod != null &&
```

Comment:
being picky, but instead of adding the null check, you could change the order of the 3 statements below so that it's implied in those, e.g. `"full".equals(ResizingImageView.resizeMethod)`